engine/src/test/java/io/graphqlcrud/SQLDataFetcherTest.java

```diff
@@ -195,6 +195,179 @@ public void testDeepNestedQuery() throws Exception {
         Assertions.assertEquals(expected,result4);
     }
 
+    @Test
+    public void simpleFilterQuery() throws Exception {
+        String query = "{\n" +
+                "  customers (filter: {\n" +
+                "    SSN: {\n" +
+                "      ne: \"CST01002\"\n" +
+                "    }\n" +
+                "  }) {\n" +
+                "    SSN\n" +
+                "    FIRSTNAME\n" +
+                "    LASTNAME\n" +
+                "  }\n" +
+                "}";
+        String result = executeSQL(query);
+        String expected = "select\n" +
+                "  \"g0\".\"SSN\" \"SSN\",\n" +
+                "  \"g0\".\"FIRSTNAME\" \"FIRSTNAME\",\n" +
+                "  \"g0\".\"LASTNAME\" \"LASTNAME\"\n" +
+                "from PUBLIC.CUSTOMER \"g0\"\n" +
+                "where \"g0\".\"SSN\" <> 'CST01002'\n" +
+                "order by \"g0\".\"SSN\"";
+        Assertions.assertEquals(expected,result);
+
+    }
+
+    @Test
+    public void nestedFilterQuery() throws Exception{
+        String query = "{\n" +
+                "  customers (filter: {\n" +
+                "    LASTNAME: {\n" +
+                "      contains: \"J\"\n" +
+                "    },\n" +
+                "    and: {\n" +
+                "      FIRSTNAME: {\n" +
+                "        startsWith: \"C\"\n" +
+                "      }\n" +
+                "    }\n" +
+                "  }) {\n" +
+                "    SSN\n" +
+                "    accounts {\n" +
+                "      ACCOUNT_ID\n" +
+                "    }\n" +
+                "  }\n" +
+                "}";
+        String result = executeSQL(query);
+        String expected = "select\n" +
+                "  \"g0\".\"SSN\" \"SSN\",\n" +
+                "  (\n" +
+                "    select json_arrayagg(json_object(key 'ACCOUNT_ID' value \"g1\".\"ACCOUNT_ID\"))\n" +
+                "    from PUBLIC.ACCOUNT \"g1\"\n" +
+                "    where \"g0\".\"SSN\" = \"g1\".\"SSN\"\n" +
+                "  ) \"accounts\"\n" +
+                "from PUBLIC.CUSTOMER \"g0\"\n" +
+                "where (\n" +
+                "  \"g0\".\"LASTNAME\" like ('%' || replace(\n" +
+                "    replace(\n" +
```

Comment:
is JOOQ doing this conversion? seems cryptic, most databases have `startsWith` and `contains` kind of functions.

Reply:
Agree, it's JOOQ. I didn't understand it as well.